src/qt/sigmacoincontroldialog.cpp

```diff
@@ -507,22 +507,62 @@ void SigmaCoinControlDialog::updateLabels(WalletModel *model, QDialog* dialog)
 
         // Priority
         dPriorityInputs += (double)out.tx->vout[out.i].nValue * (out.nDepth+1);
-
-        // Bytes
-        nBytesInputs += 74;
+        if(fMintTabSelected){
+            // Bytes
+            CTxDestination address;
+            int witnessversion = 0;
+            std::vector<unsigned char> witnessprogram;
+            if (out.tx->vout[out.i].scriptPubKey.IsWitnessProgram(witnessversion, witnessprogram))
+            {
+                nBytesInputs += (32 + 4 + 1 + (107 / WITNESS_SCALE_FACTOR) + 4);
+                fWitness = true;
+            }
+            else if(ExtractDestination(out.tx->vout[out.i].scriptPubKey, address))
+            {
+                CPubKey pubkey;
+                CKeyID *keyid = boost::get<CKeyID>(&address);
+                if (keyid && model->getPubKey(*keyid, pubkey))
+                {
+                    nBytesInputs += (pubkey.IsCompressed() ? 148 : 180);
+                    if (!pubkey.IsCompressed())
+                        nQuantityUncompressed++;
+                }
+                else
+                    nBytesInputs += 148; // in all error cases, simply assume 148 here
+            }
+            else nBytesInputs += 148;
+        }
     }
 
     // calculation
     if (nQuantity > 0)
     {
-        // Bytes         //1323 is script size for sigma spend
-        nBytes = nBytesInputs + vCoinControl.size() * 1323 +  10;
+        // Bytes
+        if(fMintTabSelected) {
+            nBytes = nBytesInputs +
+                     ((SigmaCoinControlDialog::payAmounts.size() > 0 ? SigmaCoinControlDialog::payAmounts.size() + 1 : 2) * 34) + 10; // always assume +1 output for change here
+            if (fWitness) {
+                // there is some fudging in these numbers related to the actual virtual transaction size calculation that will keep this estimate from being exact.
+                // usually, the result will be an overestimate within a couple of satoshis so that the confirmation dialog ends up displaying a slightly smaller fee.
+                // also, the witness stack size value value is a variable sized integer. usually, the number of stack items will be well under the single byte var int limit.
+                nBytes += 2; // account for the serialized marker and flag bytes
+                nBytes += nQuantity; // account for the witness byte that holds the number of stack items for each input.
+            }
+        } else {
+            // 1363 is the size of each vin containing sigma proof, 34 is the size of each normal vout,  10 is the size of empty transaction,
+            nBytes = vCoinControl.size() * 1363 + SigmaCoinControlDialog::payAmounts.size() * 34  + 10;
```

Comment:
agreed with first part of the comment, resolved. But for  second part not at all. Yes it will work for line 558 but it will cause more issues for the case when subtract fee from amount is unchecked.  as we on this step have no information about fee than we don't no actual count of remints. I think for that better to leave as it is.